helix-core/src/history.rs

```diff
@@ -133,6 +133,18 @@ impl History {
         Some(&self.revisions[last_child.get()].transaction)
     }
 
+    pub fn last_edit_pos(&mut self) -> Option<usize> {
+        let current_revision = &self.revisions[self.current];
+        if let Some(op) = &current_revision.transaction.changes().changes.first() {
```

Comment:
Hmm so this is why I didn't implement this yet, first change is not necessarily what we want.

Reply:
Though I guess we do store the original selection for the revert. Use `current_revision.inversion`'s `primary_index` to find the matching change (if `current_revision.selection.len() == changes.len()`, otherwise fallback onto `.first`)